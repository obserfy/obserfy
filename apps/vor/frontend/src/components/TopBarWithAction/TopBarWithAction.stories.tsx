import { Meta, Story } from "@storybook/react"
import TopBarWithAction, { TopBarWithActionProps } from "./TopBarWithAction"

export default {
  title: "Core/TopBarWithAction",
  component: TopBarWithAction,
  parameters: {
    componentSubtitle: "Just a simple TopBarWithAction",
  },
} as Meta

const Template: Story<TopBarWithActionProps> = (args) => (
  <TopBarWithAction {...args} />
)

export const Default = Template.bind({})
Default.args = {
  breadcrumbs: [],
}
