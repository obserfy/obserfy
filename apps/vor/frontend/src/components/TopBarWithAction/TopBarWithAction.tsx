import { t, Trans } from "@lingui/macro"
import { FC } from "react"
import { Button, Flex } from "theme-ui"
import { borderBottom } from "../../border"
import LoadingIndicator from "../LoadingIndicator/LoadingIndicator"
import TopBar, { BreadcrumbData } from "../TopBar/TopBar"
import TranslucentBar from "../TranslucentBar/TranslucentBar"

export interface TopBarWithActionProps {
  breadcrumbs: BreadcrumbData[]
  disableAction?: boolean
  onActionClick: () => void
  actionText?: string
  isLoading?: boolean
}

const TopBarWithAction: FC<TopBarWithActionProps> = ({
  disableAction,
  breadcrumbs,
  onActionClick,
  actionText = t`Add`,
  children,
  isLoading,
}) => (
  <TranslucentBar boxSx={{ position: "sticky", top: 0, ...borderBottom }}>
    <Flex sx={{ alignItems: "center", maxWidth: "maxWidth.sm" }} mx="auto">
      <TopBar breadcrumbs={breadcrumbs} />
      <Button
        data-cy="finish-cta"
        ml="auto"
        mr={3}
        px={2}
        onClick={onActionClick}
        disabled={disableAction}
      >
        {isLoading ? (
          <LoadingIndicator color="onPrimary" />
        ) : (
          <Trans id={actionText} />
        )}
      </Button>
    </Flex>

    {children}
  </TranslucentBar>
)

export default TopBarWithAction
