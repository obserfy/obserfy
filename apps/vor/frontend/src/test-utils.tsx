/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
