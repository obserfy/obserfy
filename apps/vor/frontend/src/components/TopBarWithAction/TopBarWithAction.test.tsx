import { render } from "../../test-utils"
import TopBarWithAction from "./TopBarWithAction"

describe("TopBarWithAction", () => {
  it("should render correctly", () => {
    const { container } = render(
      <TopBarWithAction
        breadcrumbs={[
          {
            text: "Test",
            to: "/",
          },
        ]}
        onActionClick={() => {}}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
