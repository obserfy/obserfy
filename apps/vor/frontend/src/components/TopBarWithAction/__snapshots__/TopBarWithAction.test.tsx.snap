// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBarWithAction should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surfaceBlurNonTransparent);
  z-index: 100;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  border-bottom-width: 1px;
  border-bottom-style: solid;
  border-bottom-color: var(--theme-ui-colors-border);
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 640px;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  margin: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  padding: 4px;
}

.emotion-5:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-5:hover,
.emotion-5:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: auto;
}

.emotion-8 {
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-8:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-8:last-child>.splitter {
  display: none;
}

.emotion-9:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-10 {
  margin-left: 4px;
  margin-right: 4px;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-onPrimary);
  background-color: var(--theme-ui-colors-primary);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: rgba(27, 31, 35, 0.1) 0px 1px 0px 0px,rgba(255, 255, 255, 0.03) 0px 1px 0px 0px inset;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-left: auto;
  margin-right: 16px;
  padding-left: 8px;
  padding-right: 8px;
}

.emotion-11:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-11:hover,
.emotion-11:focus {
  background-color: var(--theme-ui-colors-primaryDark);
}

.emotion-11:hover:disabled,
.emotion-11:focus:disabled {
  background-color: var(--theme-ui-colors-primary);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <a
            aria-current="page"
            class="emotion-4"
            data-cy="back-button"
            href="/"
          >
            <button
              class="emotion-5"
            >
              <div
                class="emotion-6"
              />
            </button>
          </a>
          <div
            class="emotion-7"
          >
            <a
              aria-current="page"
              class="emotion-8"
              href="/"
            >
              <span
                class="emotion-9"
              >
                Test
              </span>
              <span
                class="splitter emotion-10"
              >
                /
              </span>
            </a>
          </div>
        </div>
        <button
          class="emotion-11"
          data-cy="finish-cta"
        >
          Add
        </button>
      </div>
    </div>
  </div>
</div>
`;
